### PR TITLE
Add pickPeaks method

### PR DESCRIPTION
**Pull request by sgibb**
This PR adds `pickPeaks,Spectra-method`.

I like to discuss some argument names:
1. We have `halfWindowSize`, `k` for window sizes, what about `hws1`, `hws2` or
`k1`, `k2`, or `halfWindowSizeMaxima` and `halfWindowSizeRefinement`?
2. We have `threshold` here and `t` in `removePeaks` what about using `threshold`
everywhere?

### CODE REVIEW

**Comment by jorainer** [comment]
Regarding `threshold`, yes, here we should change `t` into `threshold`. For `halfWindowSize` and `k` I'm not yet sure...

File: R/Spectra.R [hunk 1]
```diff
@@ -377,6 +397,10 @@ NULL
 #'     of each spectrum in `object`. See section *Data manipulations* below
 #'     for more details.
 #'
+#' @param halfWindowSize For `pickPeaks`: `integer(1)`, a local maximum has to
+#'     be the maximum in the window from
+#'     `(i - halfWindowSize):(i + halfWindowSize)`.
```
[the remark this answers is not in the document]
Comment (sgibb) [reply R/Spectra.R]: Yes.

File: R/Spectra.R [hunk 2]
```diff
@@ -418,6 +450,10 @@ NULL
 #' @param processingQueue For `Spectra`: optional `list` of
 #'     [ProcessingStep-class] objects.
 #'
+#' @param SNR For `pickPeaks`: `double(1)` defining the
```
[the remark this answers is not in the document]
Comment (sgibb) [reply R/Spectra.R]: You are right.

File: R/Spectra.R [hunk 3]
```diff
@@ -430,6 +466,10 @@ NULL
 #'
 #' @param t for `removePeaks`: a `numeric(1)` defining the threshold or `"min"`.
```
[the remark this answers is not in the document]
Comment (sgibb) [reply R/Spectra.R]: Will change that.

File: R/Spectra.R [hunk 4]
```diff
@@ -321,6 +321,23 @@ NULL
 #'   0-intensity peaks next to non-zero intensity peaks are retained while with
 #'   `all = TRUE` all 0-intensity peaks are removed.
 #'
+#' - `pickPeaks`: picks peaks on individual spectra. For noisy spectra there
```
[the remark this answers is not in the document]
Comment (sgibb) [reply R/Spectra.R]: Right.

**Pull request merged**